### PR TITLE
Persistent Bloom Filter

### PR DESCRIPTION
**Pull request by ejfitzgerald**

### CODE REVIEW

File: cmake/BuildTools.cmake [hunk 1]
```diff
@@ -246,10 +246,18 @@ function (add_fetch_gbench
 
     # detect if the "DISABLED" flag has been passed to this test
     set(is_disabled FALSE)
+    set(include_gtest FALSE)
```
Comment (n-hutton) [inline cmake/BuildTools.cmake:249]: is this related/relevant?

File: libs/vm-modules/benchmark/CMakeLists.txt [hunk 2]
```diff
@@ -10,5 +10,11 @@ include(${FETCH_ROOT_CMAKE_DIR}/BuildTools.cmake)
 # Compiler Configuration
 setup_compiler()
 
-add_fetch_gbench(benchmark_vm_modules_model fetch-vm-modules ../../vm-modules/benchmark/model)
-add_fetch_gbench(benchmark_vm_modules_tensor fetch-vm-modules ../../vm-modules/benchmark/tensor)
+add_fetch_gbench(benchmark_vm_modules_model
+                 fetch-vm-modules
+                 ../../vm-modules/benchmark/model
+                 GTEST)
```
Comment (n-hutton) [inline libs/vm-modules/benchmark/CMakeLists.txt:16]: what is going on here?

**Review by n-hutton — DISMISSED** [review]

File: cmake/BuildTools.cmake (hunk 1 above)
Comment (ejfitzgerald) [reply cmake/BuildTools.cmake:249]: Needed in order to build the benchmarks annoyingly

File: libs/bloom-filter/include/bloom_filter/metadata_store.hpp [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+{
+  uint64_t version;
+  uint64_t length;
+};
+
+template <typename T>
+meta::IfIsPod<T> WriteToStream(std::ostream &stream, T const &element)
+{
+  auto const *raw = reinterpret_cast<char const *>(&element);
+  stream.write(raw, sizeof(T));
```
Comment (wkaluza) [inline libs/bloom-filter/include/bloom_filter/metadata_store.hpp:39]: Shouldn't this check the stream's `badbit`?

File: libs/bloom-filter/include/bloom_filter/metadata_store.hpp [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+}
+
+template <typename T>
+meta::IfIsPod<T> ReadFromStream(std::istream &stream, T &element)
+{
+  auto *raw = reinterpret_cast<char *>(&element);
+  stream.read(raw, sizeof(T));
```
Comment (wkaluza) [inline libs/bloom-filter/include/bloom_filter/metadata_store.hpp:46]: As above, this operation can fail.

File: libs/bloom-filter/include/bloom_filter/metadata_store.hpp (hunk 3 above)
Comment (ejfitzgerald) [reply libs/bloom-filter/include/bloom_filter/metadata_store.hpp:39]: Good spot - This whole of this code is temporary at the moment, actually. In case of failure it would be detected by other means. And since we rebuild the whole of the bloom filter at the moment I am inclined not to fix this here at this time.

The next stage here will be for @n-hutton to integrate his changes on and remove this metdata file.

File: libs/bloom-filter/include/bloom_filter/metadata_store.hpp (hunk 4 above)
Comment (ejfitzgerald) [reply libs/bloom-filter/include/bloom_filter/metadata_store.hpp:46]: See above 😁
